### PR TITLE
feat(context-diagrams): Add `display_parent_relation` feature

### PR DESCRIPTION
**Pull request by ewuerger**
This moves the centerbox into its parent box if it is in the context. Also other boxes are moved into the parent box if the parent-child relation holds.

**_This PR depends on [py-capellambse #381](https://github.com/DSD-DBS/py-capellambse/pull/381)._**

### CODE REVIEW

File: capellambse_context_diagrams/collectors/default.py [hunk 1]
```diff
@@ -58,10 +67,10 @@ def collector(
             makers.PORT_PADDING
             + (makers.PORT_SIZE + makers.PORT_PADDING) * len(local_ports),
         )
-        if box := global_boxes.get(i.uuid):
+        if box := global_boxes.get(i.uuid):  # type: ignore[assignment]
```
Comment (ewuerger) [inline capellambse_context_diagrams/collectors/default.py:70]: box will never be `None`. The if/else condition makes sure. Mypy not smart enough to figure it out. :(

File: capellambse_context_diagrams/serializers.py [hunk 2]
```diff
@@ -202,7 +202,7 @@ class type that stores all previously named classes.
                 pos += self.diagram[self._diagram.target.uuid].pos
 
             element = diagram.Circle(
```
Comment (huyenngn) [inline capellambse_context_diagrams/serializers.py:204]: Mypy complains
```
element: diagram.Box | diagram.Edge | diagram.Circle
```

File: capellambse_context_diagrams/collectors/default.py [hunk 3]
```diff
@@ -45,11 +45,20 @@ def collector(
...
+        del data["children"][0]
+        global_boxes[diagram.target.parent.uuid] = box
+
     stack_heights: dict[str, float | int] = {
         "input": -makers.NEIGHBOR_VMARGIN,
         "output": -makers.NEIGHBOR_VMARGIN,
     }
-    global_boxes = {centerbox["id"]: centerbox}
     child_boxes: list[_elkjs.ELKInputChild] = []
     for i, local_ports, side in port_context_collector(ex_datas, ports):
```
Comment (huyenngn) [inline capellambse_context_diagrams/collectors/default.py:63]: I feel like "i" is a bit confusing when it's not supposed to mean "index".

File: capellambse_context_diagrams/collectors/default.py [hunk 4]
```diff
@@ -75,6 +84,20 @@ def collector(
...
+            elif i.parent == diagram.target.parent:
+                parent_box = global_boxes[i.parent.uuid]
+                parent_box.setdefault("children", []).append(
+                    global_boxes.pop(i.uuid)
+                )
+                _move_edge_to_local_edges(
+                    parent_box, connections, local_ports, diagram, data
+                )
+
         stack_heights[side] += makers.NEIGHBOR_VMARGIN + height
```
Comment (huyenngn) [inline capellambse_context_diagrams/collectors/default.py:102]:
```suggestion
                edges_to_remove += _move_edge_to_local_edges(
                    box, connections, local_ports, diagram, data
                )
            elif i.parent == diagram.target.parent:
                parent_box = global_boxes[i.parent.uuid]
                parent_box.setdefault("children", []).append(
                    global_boxes.pop(i.uuid)
                )
                edges_to_remove += _move_edge_to_local_edges(
                    parent_box, connections, local_ports, diagram, data
                )

        stack_heights[side] += makers.NEIGHBOR_VMARGIN + height
    
    data["edges"] = [
        e for e in data["edges"] if e["id"] not in edges_to_remove
    ]
```

O(N*E) vs O(E) I think?

Comment (ewuerger) [reply capellambse_context_diagrams/collectors/default.py:102]: Still need to call `_move_edge_to_local_edges` inside the for-loop. With ur suggestion we'd have O((N + 1)*E).

**Pull request merged**